bundles/org.openhab.binding.huesync/src/main/java/org/openhab/binding/huesync/internal/connection/HueSyncConnection.java

```diff
@@ -0,0 +1,222 @@
+import org.openhab.binding.huesync.internal.api.dto.registration.HueSyncRegistration;
+import org.openhab.binding.huesync.internal.api.dto.registration.HueSyncRegistrationRequest;
+import org.openhab.binding.huesync.internal.log.HueSyncLogFactory;
+import org.openhab.core.io.net.http.TlsTrustManagerProvider;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.FrameworkUtil;
+import org.osgi.framework.ServiceRegistration;
+import org.slf4j.Logger;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+/**
+ * Handles the connection to a Hue HDMI Sync Box using the official API.
+ * 
+ * @author Patrik Gfeller - Initial Contribution
+ */
+public class HueSyncConnection {
+    /**
+     * Request format: The Sync Box API can be accessed locally via HTTPS on root
+     * level (port 443, /api/v1), resource level /api/v1/<resource> and in some
+     * cases subresource level /api/v1/<resource>/<subresource>.
+     */
+    private static final String REQUEST_FORMAT = "https://%s:%s/%s/%s";
+    private static final String API = "api/v1";
+    private static final Integer TIMEOUT_MILLISECONDS = 500;
+
+    private static class ENDPOINTS {
+        public static final String DEVICE = "device";
+        public static final String REGISTRATIONS = "registrations";
+    }
+
+    private static final ObjectMapper ObjectMapper = new ObjectMapper();
+
+    private final Logger logger = HueSyncLogFactory.getLogger(HueSyncConnection.class);
+
+    private @NonNull HttpClient httpClient;
+    private @NonNull String host;
+    private @NonNull Integer port;
+
+    private @Nullable String apiAccessToken;
+    private @Nullable String registrationId;
+
+    private ServiceRegistration<?> tlsProviderService;
+
+    @NonNullByDefault
+    public HueSyncConnection(HttpClient httpClient, String host, Integer port, String apiAccessToken,
+            String registrationId) throws CertificateException, IOException {
+
+        this.host = host;
+        this.port = port;
+        this.apiAccessToken = apiAccessToken;
+        this.registrationId = registrationId;
+
+        HueSyncTrustManagerProvider trustManagerProvider = new HueSyncTrustManagerProvider(this.host, this.port);
+        BundleContext context = FrameworkUtil.getBundle(getClass()).getBundleContext();
+
+        this.tlsProviderService = context.registerService(TlsTrustManagerProvider.class.getName(), trustManagerProvider,
+                null);
+        this.httpClient = httpClient;
+    }
+
+    public @Nullable HueSyncDeviceInfo getDeviceInfo() {
+        HueSyncDeviceInfo deviceInfo = null;
+
+        try {
+            ContentResponse response = this.executeGetRequest(ENDPOINTS.DEVICE);
+
+            if (response.getStatus() == HttpStatus.OK_200) {
+                deviceInfo = this.deserialize(response.getContentAsString(), HueSyncDeviceInfo.class);
```

Comment:
Honestly I am not sure why this line throws 'NoClassDefFoundError'. But I suggest that rather than doing everything in one line, you should use "defensive coding" such as the following. And also ("defensive coding") you should add 'NoClassDefFoundError' to the catch clause and log it.

```
try {
    String jsonResponse = response.getContentAsString();
    JSONObject test new JSONObject(jsonResponse);
    deviceInfo = this.deserialize(jsonResponse, HueSyncDeviceInfo.class)
} catch (JSONException | InterruptedException | ExecutionException | TimeoutException | JsonProcessingException e) {
    this.logger.error("{}", e.getMessage());
}
```